src/allmydata/scripts/runner.py

```diff
@@ -194,7 +194,30 @@ def run():
     # doesn't return: calls sys.exit(rc)
     task.react(_run_with_reactor)
 
+
+def _setup_coverage(reactor):
+    """
+    Arrange for coverage to be collected if the 'coverage' package is
+    installed
+    """
+    try:
+        import coverage
+    except ImportError:
+        return
+
+    cov = coverage.Coverage()
+    cov.start()
+
+    def write_coverage_data(*args, **kw):
+        cov.stop()
+        cov.save()
+    reactor.addSystemEventTrigger('after', 'shutdown', write_coverage_data)
```

Comment:
Somehow one test fails on CI at this line:

```
Traceback (most recent call last):
testtools.testresult.real._StringException: Traceback (most recent call last):
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
    result = f(*args, **kw)
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/twisted/internet/utils.py", line 201, in runWithWarningsSuppressed
    reraise(exc_info[1], exc_info[2])
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/twisted/internet/utils.py", line 197, in runWithWarningsSuppressed
    result = f(*a, **kw)
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/allmydata/test/cli/test_cli.py", line 541, in test_exception_catcher
    patcher.runWithPatches(runner.run)
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/twisted/python/monkey.py", line 73, in runWithPatches
    return f(*args, **kw)
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/allmydata/scripts/runner.py", line 195, in run
    task.react(_run_with_reactor)
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/allmydata/test/cli/test_cli.py", line 528, in fake_react
    d = f("reactor")
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/allmydata/scripts/runner.py", line 219, in _run_with_reactor
    _setup_coverage(reactor)
  File "/tmp/tahoe-lafs.tox/py27-coverage/local/lib/python2.7/site-packages/allmydata/scripts/runner.py", line 214, in _setup_coverage
    reactor.addSystemEventTrigger('after', 'shutdown', write_coverage_data)
AttributeError: 'str' object has no attribute 'addSystemEventTrigger'
```

This seems crazy but it looks like it's just due to a unit test doing something dumb and broken.  Perhaps `_run_with_reactor` could get a docstring along with that test abuse being cleaned up.

Reply:
The test was passing `"reactor"` (yes, a string) as the reactor, so .. I just changed it to `Mock()`. Still not "a reactor" but..